### ISSUE TITLE
Suggest: add copy files options in schema

### ISSUE THREAD

**Issue by cipchk** [ng-packagr/ng-packagr#1092]
## Type of Issue

```
[ ] Bug Report
[x] Feature Request
```

## Description

Copy `README.md` file by default. But when have multiple language versions, like this:

```
README.md
README.zh-CN.md
```

I need to manually copy other language versions. 

Sometimes, project need dependencies original styles files (like `less`) in release package.

**Comment by StudioSpindle** [comment]
Possible duplicate of #857

**Comment by Splaktar** [comment]
This doesn't seem like a duplicate as the previous was reporting that the feature didn't work. But this one is suggesting that the feature should exist.

If we want SCSS support as well, should another feature request be opened or is it appropriate to discuss that here?

**Comment by linnenschmidt** [comment]
Hi there, earlier this week I realized that there is no copy functionality for library assets.

Therefore I have extended the @angular-devkit/build-ng-packagr:build and created an own one with asset copy functionality.

Here you will find how to use it: https://github.com/linnenschmidt/build-ng-packagr#how-to-use-it

```bash
npm install @linnenschmidt/build-ng-packagr --save-dev
```
Please give it a try. Maybe this approach is enough for you.

**Comment by shlomiassaf** [comment]
Lol, I have something similar as well, just more broad...

I have also built a replacement for `@angular-devkit/build-ng-packagr:build` called [ng-cli-packagr-tasks](https://github.com/shlomiassaf/ng-cli-packagr-tasks) that provide hooks to customise `ng-packagr` at various points in the build process.

One of the examples I show is how to [copy assests](https://github.com/shlomiassaf/ng-cli-packagr-tasks/blob/master/examples/copy-files.ts)

It's quite simple, but it will be in code because it supports a wide range of things (compile css, filter builds, etc..)

**Comment by ferried** [comment]
For example. I want to publish my less. But the `ng-package.json`  has no option can do this.
It look's like copy files options: copy `*.less` to the `dest` file.
I need this option

**Comment by redplane** [comment]
Any updates about this ? It is very time wasting when I have to copy all the asset files to dist folder.

**Comment by Splaktar** [comment]
@redplane you can setup NPM tasks to do this in order to save time. Here's an [example](https://github.com/DevIntent/dev/blob/master/package.json#L16-L17).

**Comment by grey280** [comment]
I'd like to bump this, too; I'm working on an Angular library, and we've got the NPM script for building and then copying assets, but it'd be much nicer if `ng build {library} --watch` could copy over resources. Their open ticket about it [says](https://github.com/angular/angular-cli/issues/11071#issuecomment-451271094) they're waiting on a fix from over here.

**Comment by dombakargy** [comment]
manually publish your assets in some server/folder, and use environment to switch file path, if dev/prod.

environment.ts -> assetsPath = "www.gcloud.com/myServer/assets/";
environment.prod.ts -> assetsPath = "./assets/";

<img [src]=" this.environment.assetsPath + 'filename.png' ">

**Comment by EoinGriffin-AI** [comment]
I figured out a workaround to the problem yesterday.
Since `ng build --watch` must be the last thing to run (unless another process is waiting some amount of time or waiting for the presence of a generated file/directory) it's necessary to copy the assets before the build step.
Add `"deleteDestPath": false` to `ng-package.json` so that it won't clear out the build folder. That way, I can copy the assets before building and they'll stick around.
However, I actually do want to clear out the build folder, so I've added another step before copying the manually delete the build directory.
Here's a before and after of my `package.json` scripts:

**Before**
```
"build-lib": "ng build common-library --watch"
"build-app-with-lib": "ng build common-library && npm run build-app"
```

**After**
```
"copy-assets-lib": "copyfiles -u 3 ./projects/common-library/src/localization/**/*.json ./dist/common-library",
"build-lib-no-watch": "rimraf dist/common-library && npm run copy-assets-lib && ng build common-library",
"build-lib": "npm run build-lib-no-watch -- --watch",
"build-app-with-lib": "npm run build-lib-no-watch && npm run build-app"
```

**Comment by mario-d-s** [comment]
@dherges @alan-agius4
What would be the best way for contributors to help move this forward? It appears @linnenschmidt is on to something with his solution and I'm about to try it out. Of course I hope to revert to the official ng-packagr build at a later stage.

@filipesilva [talks about complexities that may not be covered](https://github.com/angular/angular-cli/issues/11071#issuecomment-451271094) with a simple solution but in my opinion a simple solution that covers some percentage of the use cases is better than none. Just wondering if @linnenschmidt's solution could be integrated as a first step and what needs to happen next.

**Comment by alan-agius4** [comment]
Hi @mario-d-s, thanks for you interest. I actually had a chat around this with @mgechev a couple of hrs ago.

We should be having more discussions around this in the near future. The first steps though would be to iron out the public api and the scope of the features, ie which scenarios will be supported and not supported.

“Copy and paste”  of assets might probably be enough for most users, and probably will be the first iteration when this is included.
